tests/test_file_plugin.py

```diff
@@ -55,15 +55,14 @@ def test_plugin_rendering(self):
         )
```

Comment:
I think we're missing a couple of tests here.

1. We should test that a draft page using the edit endpoint: https://github.com/divio/django-cms/blob/release/4.0.x/cms/toolbar/utils.py#L122 shows a draft file. This will require a few files that have different combinations, can be reused so may be worth adding them to setup:
 A. A file that is draft only
 B. A file that is published
 C. A file that has a published and draft version (draft is shown)

2. A page that is published and using the url as per a normal user in the CMS, can use the api.create_page to make this a real test I guess. Of the same files above A, B, C all should show the published variant or nothing at all on a published page.

I don't feel confident currently in this package for the above scenario and I never have. An integration / unit test that tests the above will give us a lot more confidence in future I think.

Reply:
These tests have been added.